src/ValueObject/DateValueObject.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme\ValueObject;
+
+use Drupal\Component\Datetime\DateTimePlus;
+
+/**
+ * Handle information about a date/date interval, as expected by the ECL.
+ */
+class DateValueObject extends ValueObjectBase implements DateValueObjectInterface {
+
+  /**
+   * Start date.
+   *
+   * @var \Drupal\Component\Datetime\DateTimePlus
+   */
+  protected $start;
+
+  /**
+   * End date.
+   *
+   * @var \Drupal\Component\Datetime\DateTimePlus
+   */
+  protected $end;
+
+  /**
+   * DateValueObject constructor.
+   *
+   * @param int $start
+   *   Start date as UNIX timestamp.
+   * @param int $end
+   *   End date as UNIX timestamp.
+   * @param string|null $timezone
+   *   Timezone string, e.g. "Europe/Brussels".
+   */
+  private function __construct(int $start, int $end = NULL, string $timezone = NULL) {
+    $this->start = DateTimePlus::createFromTimestamp($start, $timezone);
+
+    if ($end !== NULL) {
+      $this->end = DateTimePlus::createFromTimestamp($end, $timezone);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function fromArray(array $parameters = []): ValueObjectInterface {
+    return new static(
+      $parameters['start'],
```

Comment:
We should check if $parameters['start'] exists.

Reply:
Fixed in upcoming commit.